tests/unit/test_web3_signers.py

```diff
@@ -211,11 +217,76 @@ def test_sign_message(mocker, good_signer, mock_account, mock_key):
     assert len(signature) == LENGTH_ECDSA_SIGNATURE_WITH_RECOVERY
 
 
-def test_sign_transaction(good_signer, mock_account):
+def test_keystore_sign_transaction(good_signer, mock_account):
     transaction_dict = assoc(TRANSACTION_DICT, 'from', value=mock_account.address)
     signed_transaction = good_signer.sign_transaction(transaction_dict=transaction_dict)
     assert isinstance(signed_transaction, HexBytes)
 
     # assert valid transaction
     transaction = Transaction.from_bytes(signed_transaction)
     assert to_checksum_address(transaction.to) == transaction_dict['to']
+
+
+def test_trezor_transaction_format():
+    web3_transaction_dict = {
+        'chainId': 1,
+        'nonce': 2,
+        'gasPrice': 2000000000000,
+        'gas': 314159,
+        'to': '0xd3CdA913deB6f67967B99D67aCDFa1712C293601',
+        'value': 12345,
+        'data': b'in that metric, kman is above reproach',
+    }
+
+    trezor_transaction = TrezorSigner._format_transaction(web3_transaction_dict)
+
+    assert trezor_transaction['chain_id'] == web3_transaction_dict['chainId']
+    assert trezor_transaction['nonce'] == web3_transaction_dict['nonce']
+    assert trezor_transaction['gas_price'] == web3_transaction_dict['gasPrice']
+    assert trezor_transaction['gas_limit'] == web3_transaction_dict['gas']
+    assert trezor_transaction['to'] == web3_transaction_dict['to']
+    assert trezor_transaction['value'] == web3_transaction_dict['value']
+    assert trezor_transaction['data'] == Web3.toBytes(HexBytes(web3_transaction_dict['data']))
```

Comment:
Do we still need this line? We don't modify the 'data' value in `__format_transation` anymore

Reply:
:heavy_check_mark: